### ISSUE TITLE
Automatically check python files codestyle

### ISSUE THREAD

**Issue by mathbunnyru** [conan-io/conan-center-index#2611]
Right now it is always asked if developers followed pep8 codestyle.
I think it should be done automatically.

There is a nice tool, which can do that and it also checks some errors: flake8 (https://flake8.pycqa.org/en/latest/).

I think it would be great if all the conanfile scripts were checked automatically in the CI.

**Comment by uilianries** [comment]
Conan team discussed about it in the past, I like the idea, but it's one more barrel for contributors when they just want to change something small and now there is an extra barrel. 
As Hooks is breaking many recipes, due the legacy in the recipes, we could discuss again of course. Pinging @jgsogo @danimtb @SSE4

**Comment by mathbunnyru** [comment]
So, I have an idea how it could be implemented gradually.

The check should be implemented in conan hook.
First, the hook will whitelist all the existing packages.
This way all the new packages will look nice.

After that we're going to fix the errors in the existing packages.
I think I might fix 50 packages in one PR.
So the idea is to only change the codestyle without changing the recipe itself.
And we will remove these packages from the whitelist.

The only thing I would like to be able to do is to have a PR for a codestyle change in 50 packages, and not to create one PR for each package - that's too difficult.

This way people can make small changes and not to fix codestyle if they don't want to.

**Comment by mathbunnyru** [comment]
From one point of view, I understand that this is just codestyle and that's quite a lot of efforts.
On the other hand, these files are so important to make everything work smoothly.
So I think it's worth to make them look nice.

**Comment by mathbunnyru** [comment]
Another option is to just use black formatter (https://github.com/psf/black) on all the python files.
This way it's as simple as running one command.
But black doesn't just check if the codestyle is good enough.

> Black is a PEP 8 compliant opinionated formatter. Black reformats entire files in place. It is not configurable. It doesn't take previous formatting into account.

https://github.com/psf/black#the-black-code-style

**Comment by uilianries** [comment]
Yeah, black is a nice option indeed. Also, I like the idea creating a list for the legacy, so we can gradually move them. Thanks for sharing your ideas!!

**Comment by Croydon** [comment]
In my opinion most recipes are _simple_ or _non-complex_ enough that the coding style does not really matter in one way or another. 

@madebr wrote a `.editorconfig` file a while ago to enforce some basic formatting rules and I think that is important. Beyond that... is it really worth the troubles?

New contributors are already battling an increasing amount of hooks, we shouldn't get too small minded here

**Comment by blackliner** [comment]
Just have a look what black does to the OpenCV recipe 🙈

**Comment by madebr** [comment]
Perhaps we need to add a `pyproject.toml` with a `tool.black` section?
